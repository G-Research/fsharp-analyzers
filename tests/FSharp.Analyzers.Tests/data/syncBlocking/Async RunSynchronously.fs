module AsyncRunSynchronously

open System.Threading.Tasks

let testAsyncRunSynchronously () =
    let computation = async { return 42 }
    let result = Async.RunSynchronously computation
    result

let testAsyncRunSynchronouslyWithTimeout () =
    let computation = async { return "hello" }
    let result = Async.RunSynchronously(computation, 1000)
    result

let testAsyncRunSynchronouslyQualified () =
    let computation = async { return true }
    let result = Microsoft.FSharp.Control.Async.RunSynchronously computation
    result