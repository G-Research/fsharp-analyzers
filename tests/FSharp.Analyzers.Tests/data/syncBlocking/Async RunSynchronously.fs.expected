GRA-SYNCBLOCK-001 | Warning | (7,17 - 7,51) | Synchronous blocking call 'Async.RunSynchronously' should be avoided. This can cause deadlocks and thread pool starvation. Consider using `let!` in a `task` or `async` computation expression. Suppress with comment including text 'synchronous blocking call allowed'. | []
GRA-SYNCBLOCK-001 | Warning | (12,17 - 12,58) | Synchronous blocking call 'Async.RunSynchronously' should be avoided. This can cause deadlocks and thread pool starvation. Consider using `let!` in a `task` or `async` computation expression. Suppress with comment including text 'synchronous blocking call allowed'. | []
GRA-SYNCBLOCK-001 | Warning | (17,17 - 17,76) | Synchronous blocking call 'Async.RunSynchronously' should be avoided. This can cause deadlocks and thread pool starvation. Consider using `let!` in a `task` or `async` computation expression. Suppress with comment including text 'synchronous blocking call allowed'. | []
